File: Criacionais/Singleton/Exemplo/singleton.js
class Singlenton {
    constructor() {
        if(Singlenton.instance == null) {
            Singlenton. instance = this;
            this.value = 0;
        }

        return Singlenton.instance;
    }

    increment() {
        this.value += 1;
        console.log(`Value: ${this.value}`)
    }
}

// Cliente
const s1 = new Singlenton();
const s2 = new Singlenton();

console.log(s1 === s2);

s1.increment();
s2.increment();

File: Criacionais/Singleton/Exemplo/singleton.py
class Singleton:
    instance = None

    def __new__(cls):
        if cls.instance is None:
            cls.instance = super(Singleton, cls).__new__(cls)
            cls.instance.value = 0
        return cls.instance

    def increment(self):
        self.value += 1
        print(f"Value: {self.value}")


# Cliente
s1 = Singleton()
s2 = Singleton()

print(s1 is s2)  # True

s1.increment()
s2.increment()